### ISSUE TITLE
Refactor select component to ensure maintainability

### ISSUE THREAD

**Issue by anotherMatze** [porsche-design-system/porsche-design-system#1079]
Scope
---
Our select component has become too complex. To ensure maintainability and test coverage we have to split it internally into smaller components, which deliver the already existing functionality.

Notes  
---
* Separate Parent from Filter and Custom Dropdown
* Refactor Test setup
* Think of using `functional components`?
* Think of have an **internal** naming strategy when using `web components` for e.g. filter and dropdown

Acceptance criteria
---
- [x] Select component is maintainable again
- [x] Features for refactored select-wrapper have to stay the same

Subtasks
---
- [x] Check if we really need to initialize a new `p-icon` on every freshly selected option
- [x] Extract `p-select-wrapper-dropdown`
- [x] Extract `p-select-wrapper-filter`
- [x] Cleanup change detection of `select` and its `options`
- [x] ~Extract keyboard events handling~
- [x] Prevent react and angular wrapper generation for `p-select-wrapper-dropdown` and `p-select-wrapper-filter`
- [x] Make dropdown arrow clickable again since it is currently on top of the select
- [x] VRT tests for `p-select-wrapper-dropdown`
- [x] VRT tests for `p-select-wrapper-filter`
- [x] Unit tests for all newly extracted utils
- [x] Cleanup e2e tests for native select
- [x] Cleanup e2e tests for custom dropdown select
- [x] Cleanup e2e tests for filter/combobox select
- [x] Make component fully screen reader compliant
- [x] Add missing unit tests
- [x] Update a11y documentation in storefront
